src/Contract.tsx

```diff
@@ -4,7 +4,9 @@ export interface NavMenuItem {
 }
 
 export interface IconProps {
-    size?: number;
+    height?: number;
+    width?: number;
```

Comment:
Majority of icons are square/circle and making it possible for them to change proportions is not a good approach, the same thing to color.
In my option, we should introduce some generic type of hierarchy to handle it, e.g.:
![image](https://user-images.githubusercontent.com/37934721/217939934-1ac64b03-2dc6-47a6-81fe-5f70915e07bc.png)

interface BaseIconProps<T> {
    size?: T;
    color?: string;
    onClick?: () => void;
}

interface IconSize {
    width: number;
    height: number;
}

export interface IconProps extends BaseIconProps<number> {
    size?: number;
    color?: string;
    onClick?: () => void;
}

export interface AdvancedIconProps extends BaseIconProps<IconSize> {
    secondaryColor?: string;
}